### ISSUE TITLE
[AArch64] Failure to use neon ops for parity calculations

### ISSUE THREAD

**Issue by RKSimon** [llvm/llvm-project#56531]
https://simd.godbolt.org/z/M4qjv6Phc

For ctpop ops, aarch64 uses the neon instructions if they're available to perform fast population count. But if this has been changed into a `ISD::PARITY` node, then it remains on the scalar unit.

```llvm
define i64 @pop(i64 %x) {
  %1 = tail call i64 @llvm.ctpop.i64(i64 %x)
  ret i64 %1
}
define i64 @parity(i64 %x) {
  %1 = tail call i64 @llvm.ctpop.i64(i64 %x)
  %2 = and i64 %1, 1
  ret i64 %2
}
declare i64 @llvm.ctpop.i64(i64 %x)
```
```s
pop: // @pop
  fmov d0, x0
  cnt v0.8b, v0.8b
  uaddlv h0, v0.8b
  fmov w0, s0
  ret
.Lfunc_end0:
parity: // @parity
  eor x8, x0, x0, lsr #32
  eor x8, x8, x8, lsr #16
  eor x8, x8, x8, lsr #8
  eor x8, x8, x8, lsr #4
  eor x8, x8, x8, lsr #2
  eor w8, w8, w8, lsr #1
  and x0, x8, #0x1
  ret
.Lfunc_end1:
```

**Comment by RKSimon** [comment]
Candidate Patch for i64/i128 cases: D130246

**Comment by RKSimon** [comment]
> Candidate Patch for i64/i128 cases: D130246

This landed at 939cf9b1bea4b5daee1d1b63860b0e958703656f - its unlikely i32 and smaller types will benefit from this
